rules/expiring-todo-comments.js

```diff
@@ -158,9 +158,37 @@ function reachedDate(past) {
 	return Date.parse(past) < Date.parse(now);
 }
 
-function tryToCoerceVersion(version) {
+function tryToCoerceVersion(rawVersion) {
+	let version = rawVersion;
+
+	// Remove leading things like `^1.0.0`, `>1.0.0`
+	const leadingNoises = [
+		'>=',
+		'<=',
+		'>',
+		'<',
+		'~',
+		'^'
+	];
+	const foundTrailingNoise = leadingNoises.find(noise => version.startsWith(noise));
+	if (foundTrailingNoise) {
+		version = version.slice(foundTrailingNoise.length);
+	}
+
+	// Get only the first member for cases such as `1.0.0 - 2.9999.9999`
+	const parts = version.split(' ');
+	if (parts.length > 1) {
+		version = parts[0];
+	}
+
+	if (semver.valid(version)) {
+		return version;
+	}
+
 	try {
-		return semver.coerce(version);
+		// Try to semver.parse a perfect match while semver.coerce tries to fix errors
+		// But coerce can't parse pre-releases.
+		return semver.parse(version) || semver.coerce(version);
 	} catch (_) {
```

Comment:
```suggestion
	} catch {
```

Reply:
can we can use this?

Reply:
> can we can use this?

<details>
<summary>As engine says we support node 8, not yet</summary>

```
~/dev/eslint-plugin-unicorn:compare-pre-release-todo-conditions lubien@joestar λ nvm install 8
Downloading and installing node v8.16.2...
Downloading https://nodejs.org/dist/v8.16.2/node-v8.16.2-linux-x64.tar.xz...
############################################################################# 100,0%
Computing checksum with sha256sum
Checksums matched!
Now using node v8.16.2 (npm v6.4.1)
~/dev/eslint-plugin-unicorn:compare-pre-release-todo-conditions lubien@joestar λ npx ava test/expiring-todo-comments.js 


  ✖ No tests found in test/expiring-todo-comments.js

  1 uncaught exception

  Uncaught exception in test/expiring-todo-comments.js


  /home/lubien/dev/eslint-plugin-unicorn/rules/expiring-todo-comments.js:192
        } catch {
                ^

  SyntaxError: Unexpected token {
```

</details>

Reply:
okay